### DISCUSSION TITLE
Adding a non-project Repository

### DISCUSSION THREAD

**Discussion by MinallW** [magit/forge#645] [Ideas]
Hello Forge Community,

I have configured forge correctly on a Gitlab and can now see issues and topics, however, the place where my team does the work and adds the issues that actually are worked on, are in a Team Board which is not a project itself.

So if the projects that I am adding are /project.git, I tried to add the repository with the issues that I need in /team-example/board, and t results in a constant pull that never finishes.

So in this board, we have issues that relate to a lot of projects, so I also wonder if you could also add this globally to all projects, that would be the best ideal and what I want to use forge for.

Thank you!

**Comment by tarsius** [comment]
It is very unlikely that I will work on this any time soon. I am currently paying of technical depth. At a later time I will try to bring Gitlab support to the same level as Github support. I don't have the time to implement Gitlab only features. Sorry.

**Reply by MinallW** [reply]
Alright, when the time is right, I'll be around to help you with any tests needs,

Thank you!

**Reply by MinallW** [reply]
Is there a way I can help you implement this feature in any way?, how complicated would be to implement this? 
(I am a programmer, not a Lisp one but I have small experience in it)

**Reply by tarsius** [reply]
> Is there a way I can help you implement this feature in any way?

I'm afraid not. A lot of the technical depth I am currently paying of is due to merging minimal viable implementations, whether implemented by users like you or by me. A lot of the bad code that I implemented myself, I only added because a user first implemented that functionality and thus demonstrated that it was "easy". Not wanting to explain why those implementations were not sufficient I implemented a somewhat better version myself. These somewhat better implementations are now turning out to be very painful to improve. I'm not doing that again.

**Comment by tarsius** [comment]
I do not intend to add support for Gitlab "boards" and Github "projects".

> Adding a non-project Repository

I think what you are asking for is that I *do* add support such "boards". I'm afraid, I don't have the bandwidth to implement that.

If, on the other hand, you are just asking to add support for adding a repository to the database, without cloning it locally (because that would be pointless when it comes to Git, because it is empty), then you will be happy to hear that it is now possible to add a repository to the Forge database, even when there is no local checkout.
